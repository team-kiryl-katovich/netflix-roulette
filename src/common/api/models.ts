export type Result<T> = {
  data?: T;
  error?: Error;
};

export type Offset = {
  offset?: number;
  limit?: number;
};

export type OffsetResult<T> = Result<T> &
  Offset & {
    total?: number;
  };

export type Movie = {
  id: number;
  title: string;
  tagline: string;
  vote_average: number;
  vote_count: number;
  release_date: string;
  poster_path: string;
  overview: string;
  budget: number;
  revenue: number;
  runtime: number;
  genres: string[];
};

export type SortBy = keyof Movie;
export type SortOrder = 'desc' | 'asc';
export type SearchBy = Extract<keyof Movie, 'title' | 'genres'>;

export type MovieFilter = {
  sortBy?: SortBy;
  sortOrder?: SortOrder;
  search?: string;
  searchBy?: SearchBy;
  filter?: string;
};
