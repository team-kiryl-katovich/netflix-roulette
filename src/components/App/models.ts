import { Store } from 'redux';

import { AppState } from '../../store/models';

export type AppProps = {
  store: Store<AppState>;
};
