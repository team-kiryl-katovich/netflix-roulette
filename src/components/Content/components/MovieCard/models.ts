export type MovieCardProps = {
  id: number;
  title: string;
  gengre: string;
  year: number;
  imageUrl: string;
};
