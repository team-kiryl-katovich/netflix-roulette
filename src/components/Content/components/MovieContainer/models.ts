import { WithMoviesProps } from 'src/enhancers/withMovies';

export type MovieContainerComponentProps = WithMoviesProps;
