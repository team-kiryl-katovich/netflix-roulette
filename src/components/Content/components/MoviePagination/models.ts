import { Movie, Offset } from '@common/api';

export type MoviePaginationProps = {
  moviesCount: number;
  offset: Offset;
  pageSize: number;
  loading: boolean;
  children?: React.ReactNode;
  fetch: (offset: Offset) => void;
};

export type MoviePaginationState = {};
