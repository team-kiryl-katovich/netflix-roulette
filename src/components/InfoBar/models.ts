import { WithFilterProps } from '@enhancers/withFilter';
import { WithMoviesProps } from '@enhancers/withMovies';

export type InfoBarComponentProps = WithFilterProps & WithMoviesProps;
