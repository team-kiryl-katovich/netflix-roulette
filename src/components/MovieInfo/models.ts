import { WithMovieProps } from '@enhancers/withMovie';

export type MovieInfoComponentProps = WithMovieProps;
