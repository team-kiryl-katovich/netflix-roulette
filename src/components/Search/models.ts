import { WithMoviesHandlersProps } from '@enhancers/withMovies';
import { WithFilterProps } from '@enhancers/withFilter';

export type SearchComponentProps = WithFilterProps & WithMoviesHandlersProps;
