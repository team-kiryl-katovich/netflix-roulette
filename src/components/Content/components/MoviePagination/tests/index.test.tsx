import * as React from 'react';
import * as enzyme from 'enzyme';

import { MoviePaginationProps } from '../models';
import { MoviePagination } from '..';

describe('[COMPONENTS] [CONTENT] [COMPONENTS] [MOVIE_PAGINATION]', () => {
  const fakeProps: MoviePaginationProps = {
    fetch: jest.fn(),
    loading: false,
    moviesCount: 0,
    offset: {
      limit: 1,
      offset: 1,
    },
    pageSize: 1,
  };

  test('should render component', () => {
    const component = enzyme.shallow(
      <MoviePagination {...fakeProps}>
        <div>fake children</div>
      </MoviePagination>
    );
    expect(component).toMatchSnapshot();
  });

  test('should scroll and unmount', () => {
    (fakeProps.fetch as jest.Mock).mockReset();
    const component = enzyme.mount(
      <MoviePagination {...fakeProps}>
        <div>fake children</div>
      </MoviePagination>
    );

    window.dispatchEvent(new window.UIEvent('scroll'));
    component.unmount();

    expect(fakeProps.fetch).toHaveBeenCalled();
  });
});
