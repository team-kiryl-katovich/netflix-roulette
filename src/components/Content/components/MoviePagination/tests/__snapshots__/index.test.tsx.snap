// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[COMPONENTS] [CONTENT] [COMPONENTS] [MOVIE_PAGINATION] should render component 1`] = `
<Fragment>
  <div>
    fake children
  </div>
  <Component
    loading={false}
  />
</Fragment>
`;
